spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/BoxControllerHelperService.java

```diff
@@ -238,7 +238,7 @@ public JSONObject removeTubeFromBox(HttpSession session, JSONObject json) {
    *          session, JSONObject json
    * @returns JSONObject message indicating failure or success
    */
-  public JSONObject discardSingleTube(HttpSession session, JSONObject json) {
+  public JSONObject discardSingleItem(HttpSession session, JSONObject json) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "session". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)